build.proj

```diff
@@ -60,12 +60,18 @@
 
   <Target Name="ValidateExactRestore"
           Condition="'$(AllowInexactRestore)'!='true'">
-    <ValidateExactRestore ProjectLockJsons="@(ProjectJsonFiles->'%(RootDir)%(Directory)%(Filename).lock.json')" />
+    <ItemGroup>
+      <ProjectJsonTemplateFiles Include="$(MSBuildThisFileDirectory)**\project.json.template" />
+      <ProjectJsonsExcludingTemplateFiles Include="@(ProjectJsonFiles)" Exclude="@(ProjectJsonTemplateFiles)" />
+    </ItemGroup>
+    <ValidateExactRestore ProjectLockJsons="@(ProjectJsonsExcludingTemplateFiles->'%(RootDir)%(Directory)%(Filename).lock.json')" />
```

Comment:
Can this be changed to support project.json.template files rather than ignore them? (We have evaluated the template and restored them at this point, right?) There is value in this validation--protection against bad auto-upgrades and manual tweaks.

On the other hand, when we get rid of the `project.json.template` files, this validation won't work, so maybe it's not worth it. I don't know how far away that is.